### PR TITLE
[ja] Updated /content/ja/docs/concepts/workloads/pods/init-containers.md file

### PR DESCRIPTION
**Pull request by ziyi-xie**
There are several updates in [the original English text](https://github.com/kubernetes/website/blob/main/content/en/docs/concepts/workloads/pods/init-containers.md) and therefore this page is outdated.
This PR updated Japanese translation for the [/content/ja/docs/concepts/workloads/pods/init-containers.md](https://kubernetes.io/ja/docs/concepts/workloads/pods/init-containers/) file.

Fixes https://github.com/kubernetes/website/issues/36515

### CODE REVIEW

**Comment by ziyi-xie** [comment]
/assign @nasa9084
/cc @atoato88 @t-inu
Hi! I've updated Japanese translation for the [/content/ja/docs/concepts/workloads/pods/init-containers.md](https://kubernetes.io/ja/docs/concepts/workloads/pods/init-containers/) file, would you mind checking if everything looks alright at your convenience time please? Here is [the original English text](https://github.com/kubernetes/website/blob/main/content/en/docs/concepts/workloads/pods/init-containers.md) and thank you so much!

File: content/ja/docs/concepts/workloads/pods/init-containers.md [hunk 1]
```diff
@@ -196,36 +203,42 @@ Initコンテナが完了し、`myapp-pod`というPodがRunning状態に移行
...
 kubectl get -f myapp.yaml
+```
+実行結果は下記のようになります。
+```
 NAME        READY     STATUS    RESTARTS   AGE
 myapp-pod   1/1       Running   0          9m
 ```
 
-このシンプルな例を独自のInitコンテナを作成する際の参考にしてください。[次の項目](#whats-next)にさらに詳細な使用例に関するリンクがあります。
+このシンプルな例を独自のInitコンテナを作成する際の参考にしてください。[次の項目](#what-s-next)にさらに詳細な使用例に関するリンクがあります。
```
Comment (nasa9084) [inline content/ja/docs/concepts/workloads/pods/init-containers.md:213]: TBH, I'm not sure where the ID comes from but the ID is `#次の項目` for now and it should be translated as well
(here is the translation for whatsnext_heading: https://github.com/kubernetes/website/blob/main/data/i18n/ja/ja.toml#L208-L209, however, I could find where it is used)

File: content/ja/docs/concepts/workloads/pods/init-containers.md [hunk 2]
```diff
@@ -196,36 +203,42 @@ Initコンテナが完了し、`myapp-pod`というPodがRunning状態に移行
...
 
-Podの起動時は各Initコンテナが起動状態となるまで、kubeletはネットワーキングおよびストレージを利用可能な状態にしません。また、kubeletはPodのspecに定義された順番に従ってPodのInitコンテナを起動します。各Initコンテナは次のInitコンテナが起動する前に正常に終了しなくてはなりません。もしあるInitコンテナがランタイムもしくはエラーにより起動失敗した場合、そのPodの`restartPolicy`の値に従ってリトライされます。しかし、もしPodの`restartPolicy`が`Always`に設定されていた場合、Initコンテナの`restartPolicy`は`OnFailure`が適用されます。
+Podの起動時に、kubeletはネットワークおよびストレージの準備が整うまで、Initコンテナを実行可能な状態にしません。また、kubeletはPodのspecに定義された順番に従ってPodのInitコンテナを起動します。
+
+各Initコンテナは次のInitコンテナが起動する前に正常に終了しなくてはなりません。もしあるInitコンテナがランタイムにより起動失敗した場合、もしくはエラーで終了した場合、そのPodの`restartPolicy`の値に従ってリトライされます。しかし、もしPodの`restartPolicy`が`Always`に設定されていた場合、Initコンテナの`restartPolicy`は`OnFailure`が適用されます。
 
 Podは全てのInitコンテナが完了するまで`Ready`状態となりません。Initコンテナ上のポートはServiceによって集約されません。初期化中のPodのステータスは`Pending`となりますが、`Initialized`という値はtrueとなります。
 
-もしそのPodが[再起動](#pod-restart-reasons)されたとき、全てのInitコンテナは必ず再度実行されます。
+もしそのPodを[再起動](#pod-restart-reasons)する、または再起動されたとき、全てのInitコンテナは必ず再度実行されます。
```
Comment (nasa9084) [inline content/ja/docs/concepts/workloads/pods/init-containers.md:223]:
```suggestion
もしそのPodを[再起動](#pod-restart-reasons)するとき、または再起動されたとき、全てのInitコンテナは必ず再度実行されます。
```

File: content/ja/docs/concepts/workloads/pods/init-containers.md [hunk 3]
```diff
@@ -196,36 +203,42 @@ Initコンテナが完了し、`myapp-pod`というPodがRunning状態に移行
...
+すでに正しく動作しているPodは`activeDeadlineSeconds`を設定すると強制終了されます。
 
 Pod内の各アプリケーションコンテナとInitコンテナの名前はユニークである必要があります。他のコンテナと同じ名前を共有していた場合、バリデーションエラーが返されます。
 
 ### リソース {#resources}
 
 Initコンテナの順序と実行を考えるとき、リソースの使用に関して下記のルールが適用されます。
 
-* 全てのInitコンテナの中で定義された最も高いリソースリクエストとリソースリミットが、*有効なinitリクエスト／リミット* になります。
+* 全てのInitコンテナの中で定義された最も高いリソースリクエストとリソースリミットが、*有効なinitリクエスト／リミット* になります。リソースリミットが設定されていない場合、最上級のリミットとみなされます。
```
Comment (nasa9084) [inline content/ja/docs/concepts/workloads/pods/init-containers.md:241]:
```suggestion
* 全てのInitコンテナの中で定義された最も高いリソースリクエストとリソースリミットが、*有効なinitリクエスト／リミット* になります。いずれかのリソースでリミットが設定されていない場合、これが最上級のリミットとみなされます。
```
no resource limitは無条件で「最も高い」と見なされる、ということだと思うんですが、正直この文だと伝わらないんじゃないかな、と思っています(が、いい訳文が思いつかない・・・ cc @kubernetes/sig-docs-ja-reviews )

File: content/ja/docs/concepts/workloads/pods/init-containers.md [hunk 4]
```diff
@@ -240,10 +253,10 @@ Podレベルのコントロールグループ(cgroups)は、スケジューラ
 
 以下の理由によりPodは再起動し、Initコンテナの再実行も引き起こす可能性があります。
 
-* ユーザーが、そのPodのInitコンテナのイメージを変更するようにPodの仕様を更新する場合。アプリケーションコンテナのイメージの変更はそのアプリケーションコンテナの再起動のみ行われます。
 * そのPodのインフラストラクチャーコンテナが再起動された場合。これはあまり起きるものでなく、Nodeに対するルート権限を持ったユーザーにより行われることがあります。
-* `restartPolicy`が`Always`と設定されているPod内の全てのコンテナが停止され、再起動が行われた場合。およびガーベージコレクションによりInitコンテナの完了記録が失われた場合。
+* `restartPolicy`が`Always`と設定されているPod内の全てのコンテナが停止され、再起動が行われるが、ガーベージコレクションによりInitコンテナの完了記録が失われた場合。
```
Comment (nasa9084) [inline content/ja/docs/concepts/workloads/pods/init-containers.md:257]:
```suggestion
* `restartPolicy`が`Always`と設定されているPod内の全てのコンテナが停止され、再起動が行われたが、ガーベージコレクションによりInitコンテナの完了記録が失われた場合。
```

File: content/ja/docs/concepts/workloads/pods/init-containers.md [hunk 5]
```diff
@@ -240,10 +253,10 @@ Podレベルのコントロールグループ(cgroups)は、スケジューラ
...
 
+initコンテナのイメージが変更されたり、ガベージコレクションによってinitコンテナの完了記録が失われた場合、Podは再起動されません。これはKubernetes v1.20以降に適用されます。以前のバージョンを使用している場合は、対応バージョンのドキュメントを参照してください。
```
Comment (nasa9084) [inline content/ja/docs/concepts/workloads/pods/init-containers.md:259]:
```suggestion
Kubernetes v1.20以降では、initコンテナのイメージが変更されたり、ガベージコレクションによってinitコンテナの完了記録が失われたりした場合、Podは再起動されません。以前のバージョンを使用している場合は、対応バージョンのドキュメントを参照してください。
```

「場合でも」とした方がいいかも？(cc @kubernetes/sig-docs-ja-reviews )

File: content/ja/docs/concepts/workloads/pods/init-containers.md (hunk 1 above)
Comment (atoato88) [reply content/ja/docs/concepts/workloads/pods/init-containers.md:213]: If we use ` [.....](#what-s-next)`, it needs to add ` {#what-s-next}` anchor markup on last section in this page , I think.
I left comment about it below.

File: content/ja/docs/concepts/workloads/pods/init-containers.md [hunk 6]
```diff
@@ -240,10 +253,10 @@ Podレベルのコントロールグループ(cgroups)は、スケジューラ
...
 
 ## {{% heading "whatsnext" %}}
```
Comment (atoato88) [inline content/ja/docs/concepts/workloads/pods/init-containers.md:261]:
```suggestion
## {{% heading "whatsnext" %}} {#what-s-next}
```
I guess we shoud add explicit anchor markup for jumping from "次の項目" link above.

Comment (atoato88) [reply content/ja/docs/concepts/workloads/pods/init-containers.md:261]: but this works correctly?

**Review by atoato88 — COMMENTED** [review]
I left comments about anchor link.

File: content/ja/docs/concepts/workloads/pods/init-containers.md (hunk 5 above)
Comment (nasa9084) [reply content/ja/docs/concepts/workloads/pods/init-containers.md:259]:
```suggestion
Kubernetes v1.20以降では、initコンテナのイメージが変更されたり、ガベージコレクションによってinitコンテナの完了記録が失われたりした場合でも、Podは再起動されません。以前のバージョンを使用している場合は、対応バージョンのドキュメントを参照してください。
```

**Comment by atoato88** [comment]
/lgtm
And I checked that `次の項目` link works correctly. :+1:

**Review by nasa9084 — COMMENTED** [review]
/approve

**Pull request merged**